### DISCUSSION TITLE
How to adjust dependent analysis results

### DISCUSSION THREAD

**Discussion by lll-lll-lll-lll** [explosion/spaCy#8905] [Help: Coding & Implementations] [answered]
Dependency parsing results were different for commas and the conjunction "and".
Please tell me how to improve the dependency parsing after "The person who uses his camera occasionally to capture a birthday party".

First image.
<img src ="https://raw.githubusercontent.com/lll-lll-lll-lll/data/4eba22448f84e0cfcff5af7cc29d7bf5b96c86ed/not_want_dep.svg" width=1500>
In the case of "a birthday party, a scenic view", "the person" and "a scenic view" are appositions.

That' s not the result I want.
The reason is that when Google Translate converts it into Japanese, the result is semantically different.

When I replaced the "and" with a comma, the result of the dependency analysis was different.
This is the second image.

<img src ="https://raw.githubusercontent.com/lll-lll-lll-lll/data/4eba22448f84e0cfcff5af7cc29d7bf5b96c86ed/want_dep.svg" width=1000>
The result is that "a birthday party", "a scenic view", and "a family outing" are connected.
This is the result I want.

Would it be reasonable to add a rule to treat commas as conjunctions?
I'm glad you can give me some advice.
Thanks.

Translated with www.DeepL.com/Translator Japanese to English (free version)

**Comment by polm** [comment, marked as answer]
> Would it be reasonable to add a rule to treat commas as conjunctions?

The dependency parser isn't rule based. The way it works it by looking at a lot of example sentences, it learns a statistical model that it uses to make decisions about how to classify things. This means that we can't "fix" or control specific instances, because everything depends on everything else. See #3052 for some details on that.

You can manually edit the dependency parse, so if your conditions are regular enough, you can write code to edit things to be the way you want. For long range dependencies like this it might be hard though.

> The reason is that when Google Translate converts it into Japanese, the result is semantically different.

I don't understand - how does spaCy's dependency parse affect Google Translate results?

**Reply by lll-lll-lll-lll** [reply]
Thanks for the reply.
I'm going to set a dependency parse.

> I don't understand - how does spaCy's dependency parse affect Google Translate results?

The google translate shows "a birthday party, a scenic view or a famliy outing" as the object of "capture". 
In the analysis of the first image, the "comma" did not work as a connection between "a birthday party" and "a scenic view"
However, when I changed the three objects to "A, B or C", for example, the result became what I wanted.

<img src="https://raw.githubusercontent.com/lll-lll-lll-lll/data/819860085e052471632b49f3c409b8c8d80e9362/want_dep.svg">
In other words, I connected them as objects.  
I will try to solve this problem by providing a rule.
